docs/deploy/index.md

```diff
@@ -0,0 +1,148 @@
+At CERN we have been running Invenio v3 instances using [OpenStack](https://www.openstack.org)
+to manage the VMs, and [Puppet](https://www.puppet.com) for IaC (Infrastructure as Code).
+Other common IaC technologies are [Ansible](https://www.ansible.com) and [Terraform](https://www.terraform.io).
+
+![Deployment models](../images/deployment-models.png)
+
+You could also keep going down in the layers to get more and more control of
+your deployment, for example, using bare metal. However, that is out of the
+scope of this documentation.
+
+!!! warning "Docker compose is not recommended"
+    The provided docker-compose files can give you a good idea on what you need
+    to deploy, including the underlying services. However, the provided files
+    are **NOT** meant to be used in production as-is: it requires knowledge
+    and experience on how to run it reliable, safely stores files, understand
+    potential performance limitations, etc. Use at your own risk.
+
+## Services
+
+When deploying InvenioRDM you can choose to deploy and maintain yourself the
+services, or use third party providers. For example, the three cloud providers
+mentioned above (AWS, GC, and Azure) can provide most of the services. If you
+choose to deploy them yourself you will need to take into account the following
+topics:
+
+- Persist your data, and enable periodic backups. This includes the relational
+database, the search indices and the files.
+- Queue persistence (RabbitMQ) to avoid loosing tasks in case the service fails.
+- High availability, many of the services can be deployed redundantly. Note
+that most of the cloud providers offer this option or have an established SLA.
+- Secrets handling. Most of the services require credentials to connect. It is
+a good practice to keep them in a secure place, for example [Ansible Vault](https://www.redhat.com/sysadmin/introduction-ansible-vault).
+
+If you choose to deploy them using PaaS, images for the services can be
+found for example in [DockerHub](https://hub.docker.com). For more information
+on how to deploy each service, refer to their official documentation.
+
+## InvenioRDM Python application
+
+Independently of your deployment model choice, you will need to run the Flask
+Python application. However, the server bundled with Flask is not meant for
+production usage. Therefore, you will need some sort of WSGI HTTP server.
+Common choices are [Gunicorn](https://gunicorn.org) and [uWSGI](http://projects.unbit.it/uwsgi),
+The latter, is the one we use at CERN and it is bundled in the Dockerfile.
+
+### Versioning
+
+During the lifetime of your service you will develop new features specific to
+your instance or need to upgrade to a newer version of InvenioRDM. This will
+means that your application code will change and versioning it will help you
+control which code is actually deployed. At CERN we use GitHub tags/release
+for each version (e.g. v1.0.0) plus a `latest` and `production` tag, these two
```

Comment:
Suggestion: I might be tempted to say to remove the `latest` and `production` tags, they can be dangerous if you don't know exactly what you are doing.

Reply:
Agree